octoprint_octorelay/__init__.py

```diff
@@ -424,10 +425,22 @@ def input_polling(self):
                 self.update_ui()
                 break
 
+    def get_additional_permissions(self):
+        return [{
+            "key": "SWITCHING",
+            "name": "Switching relays ON and OFF",
+            "description": "Allows to toggle the GPIO pins and execute OS commands associated with them",
+            "roles": [ "default" ],
```

Comment:
I'm not sure what this one should be.
Documentation here: https://docs.octoprint.org/en/master/plugins/hooks.html#octoprint-access-permissions

Reply:
Here is how to use it:

```
    def on_api_get(self, request):
        if not Permissions.PLUGIN_{CODE}_{KEY}.can():
            return flask.abort(403)
```

https://github.com/OctoPrint/OctoPrint-PiSupport/blob/7fcdeaec0cb663523942dd017e25a79a3daaab69/octoprint_pi_support/__init__.py#L398C1-L400C36